### DISCUSSION TITLE
Does iTwin Shapshot support Civil 3D 2024 files?

### DISCUSSION THREAD

**Discussion by JohnMajerle** [iTwin/itwinjs-core#6227] [Q&A]
I have a customer that uses Civil 3D.  I am evaluating whether or not we can develop a usedful iTwin workflow for them. 

Using iTwin Shapshot I created a bim file from a sample dwg file delivered with Civil 3D 2024.  If you install that product it can be found at:
`C:\Program Files\Autodesk\AutoCAD 2024\C3D\Sample\Civil 3D API\DotNet\CSharp\Spiral2 Demo\demo.dwg`

The image below is a side by side comparison of demo.dwg and the resultant bim file viewed from the iTwin Snapshot viewer.  Notice that hardly any of the dwg elements ended up in the bim, and many of the elements in that file are nonsensical.  

Does iTwin support Civil 3D conversion?  If so, what are the limitations?

![image](https://github.com/iTwin/itwinjs-core/assets/30668739/667bd4f9-c1c8-4bc5-88bb-5f3e60a0631f)

**Comment by JohnMajerle** [comment]
I'm still waiting for an answer on this.

**Reply by donhfu** [reply]
Did you pick AutoCAD or Civil3d connector from your Snapshot? You can confirm that by checking an element name - does it say "ACAD_PROXY_ENTITY"?

You can use AutoCAD connector for this file, so long as you don't care about Civil data and you have Autodesk's Civil3d object enabler v2022 installed on the server you run the connector.  The output bim will have geometry enabled by the Civild3d OE, but won't have the Civil schema.

Civil3d connector is recommended for this file.  Maybe you can give it shot?

**Comment by jffmarker** [comment]
@donhfu can you please comment on the expectations of c3d conversion?

As a snapshot, I would not expect any reality data to come across... you'll need to use a cloud-based iTwin for that. I'm not sure off-hand what to expect for the other elements though.

CC: @timlawrence-bentley
